src/watch_filter.rs

```diff
@@ -17,49 +12,64 @@ use watchexec::{
     signal::source::MainSignal::Interrupt,
 };
 
+/// Filters Watchexec events.
+///
+/// Also enables checking whether an application manifest, source, or artifact was modified.
 #[derive(Debug)]
-pub struct WatchFilter {
-    path_patterns: Vec<Pattern>,
-    ignore_patterns: Vec<Pattern>,
+pub struct Filter {
```

Comment:
Consider keeping a mention of "watch" in the name.

Reply:
Since I'm planning to move this to it's own crate I'm going to keep the name as just `Filter` since it is implied that it is for watching -- given the crate.